src/org/skypro/skyshop/search/SearchEngine.java

```diff
@@ -0,0 +1,32 @@
+package org.skypro.skyshop.search;
+
+public class SearchEngine {
+    private final Searchable[] searchableItems;
+    private int count = 0;
+
+    public SearchEngine(int size) {
+        searchableItems = new Searchable[size];
+    }
+
+    public void add(Searchable item) {
+        if (count < searchableItems.length) {
+            searchableItems[count++] = item;
+        }
+    }
+
+    public Searchable[] search(String term) {
+        Searchable[] results = new Searchable[5];
+        int resultCount = 0;
+
+        for (Searchable item : searchableItems) {
+            if (item != null && item.searchTerm().contains(term)) {
+                if (resultCount < 5) {
```

Comment:
используем длину массива, а не число с потолка
```suggestion
                if (resultCount < results.length) {
```